cmd/up.go

```diff
@@ -142,3 +165,46 @@ func isGKE(disco discovery.DiscoveryInterface) (bool, error) {
 
 	return false, nil
 }
+
+func mongoSecretExists(c kubecfg.ApplyCmd, name, ns string) (*unstructured.Unstructured, bool, error) {
+	gvk := schema.GroupVersionKind{Version: "v1", Kind: "Secret"}
+	rc, err := clientForGroupVersionKind(c.ClientPool, c.Discovery, gvk, ns)
+	if err != nil {
+		return nil, false, err
+	}
+	pwFields := []string{"mongodb-password", "mongodb-root-password"}
+	prevSec, err := rc.Get(name)
+	if err != nil {
+		if k8sErrors.IsNotFound(err) {
+			return nil, false, nil
+		} else {
+			return nil, true, err
+		}
+	} else {
+		if prevSec.Object["data"] != nil {
+			prevPw := prevSec.Object["data"].(map[string]interface{})
+			for _, p := range pwFields {
+				if prevPw[p] == nil {
+					return nil, true, fmt.Errorf("secret %s already exists but it doesn't contain the expected key %s", name, p)
+				}
+			}
+		} else {
```

Comment:
Invert return-early if/else clauses like this to keep a linear flow (again, readability).  Compare:

```golang
if prevSec.Object["data"] == nil {
    return nil, true, fmt.Errorf("secret already exists but  ...")
}
prevPw := prevSec.Object["data"].(map[string]interface{})
for _, p := range pwFields {
    if prevPw[p] == nil {
        return nil, true, fmt.Errorf("secret doesn't contain key ...")
    }
}
return prevSec, true, nil
```

Reply:
👍